lib/http.js

```diff
@@ -48,6 +49,13 @@ SwaggerHttp.prototype.execute = function (obj, opts) {
 
   var success = obj.on.response;
 
+  var requestInterceptor = function(data) {
+    if(opts && opts.requestInterceptor) {
+      data = opts.requestInterceptor.apply(data);
+    }
+    return data;
+  }
+
   var responseInterceptor = function(data) {
     if(opts && opts.responseInterceptor) {
       data = opts.responseInterceptor.apply(data);
```

Comment:
How would I get access to the `req`, `res`, and `err` objects to support #599 ???